pkg/scheduler/queue/tree_queueing_algorithms.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package queue
+
+// DequeueAlgorithm represents the set of operations specific to different approaches to dequeuing. It is applied
+// at the layer-level -- every Node at the same depth in a TreeQueue shares the same DequeueAlgorithm, including state
+// that may be stored in a struct that implements DequeueAlgorithm.
+type DequeueAlgorithm interface {
+	addChildNode(parent, child *Node)
+	dequeueSelectNode(node *Node) (*Node, bool)
+	dequeueUpdateState(node *Node, dequeuedFrom *Node)
```

Comment:
I think it would help to write down what each of these is for. As is these don't feel like a cohesive set of methods related to "dequeue"-ing as the name would imply. I'd find it hard to implement this interface in the future if needed.

Reply:
+1